lib/realtime/headway.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Realtime.Headway do
+  alias Gtfs.{Direction, Route, Stop}
+  alias Realtime.TimePeriod
+
+  @type headway_spacing ::
+          :very_bunched
+          | :bunched
+          | :ok
+          | :gapped
+          | :very_gapped
+
+  @type key_route_headways :: %{Route.id() => direction_origin_headways()}
+
+  @type direction_origin_headways :: %{Direction.id() => origin_headways()}
+
+  @type origin_headways :: %{Stop.id() => [time_period_headway()]}
+
+  @type time_period_headway :: %{
+          time_period_name: String.t(),
+          average_headway: String.t() | nil
+        }
+
+  @type seconds :: non_neg_integer()
+
+  @spec current_headway_spacing(seconds(), seconds()) :: headway_spacing()
+  def current_headway_spacing(headway_seconds, expected_headway_seconds)
```

Comment:
this is again subjective, but if you swapped these arguments you'd be able to do 

```
spacing =
  route_id
  |> Headway.current_expected_headway_seconds(
    direction_id,
    origin_stop_id,
    date_time_now_fn.()
  )
  |> Headway.current_headway_spacing()


```